docs/advanced-topics/networkobject-parenting.md

```diff
@@ -60,27 +45,52 @@ If you run into a situation where your client must trigger parenting a `NetworkO
+A `NetworkObject` can only be parented under another `NetworkObject`. The only exception is if you don't want the `NetworkObject` to have any parent. Under this case, you would parent to the root of the scene hierarchy (i.e. setting the `transform.parent` to `null`).  
 
-:::info
-The `NetworkObject` requirement is primarily for identification purposes (i.e. knowing which GameObject's transform we are going to parent under).
+:::tip
+You can use `NetworkObject.RemoveParent` to remove a parent from a child which doesn't require you to cast `null` to a `GameObject` or `NetworkObject` when using `NetworkObject.TrySetParent`.  This helper method was added for convenience and internally it does the casting for you:
+ ```csharp
+ public bool TryRemoveParent(bool worldPositionStays = true)
+ {
+     return TrySetParent((NetworkObject)null, worldPositionStays);
+ }
+```
 :::
 
 ### Only Spawned NetworkObjects Can Be Parented
 A `NetworkObject` can only be parented if it is spawned and can only be parented under another spawned `NetworkObject`. This also means that `NetworkObject` parenting can only occur during a network session (netcode enabled game session).  Think of `NetworkObject` parenting as a netcode event.  In order for it to happen, you must have, at very minimum, a server or host instance started and listening.
 
 ### Invalid `NetworkObject` Parenting Will Be Reverted
-If an invalid/unsupported `NetworkObject` parenting happens, Netcode will immediately revert it back to its original parenting status. 
+If an invalid/unsupported `NetworkObject` parenting happens, the attempted parenting action will be immediately reverted it back to its original parenting status. 
 
 **For example:** 
 If you had a `NetworkObject` who's current parent was root and tried to parent it in an invalid way (i.e. under a GameObject without a `NetworkObject` component) then a warning message would be logged and the `NetworkObject` would revert back to having root as its parent.
 
 ### In-scene NetworkObject parenting of players
-In-scene placed `NetworkObject` parenting of players requires the client to be synchronized first.  Since a server can only perform parenting related actions, the server must have already received the `NetworkSceneManager` generated `SceneEventType.SynchronizeComplete` message before the server can parent the client's player `NetworkObject`.
+In-scene placed `NetworkObject` parenting of players requires the client to be synchronized first.  Since a server can only perform parenting related actions, the server must have already received the `NetworkSceneManager` generated `SceneEventType.SynchronizeComplete` message  before the server can parent the client's player `NetworkObject`.
```

Comment:
You must synchronize clients before parenting players with in-scene placed `NetworkObject`. Because a server can only perform parenting-related actions, the server must have already received the `NetworkSceneManager` generated `SceneEventType.SynchronizeComplete` message before the server can parent the client's player `NetworkObject`.

I'm not sure about the correctness of this revision. I leave it to your expertise 😄 .

Reply:
Uh...yeah that whole section I still am not happy with.

Basically this is what I am trying to convey:
Since in-scene placed NetworkObjects are spawned after the scene they are placed within has loaded, parenting of players requires that the clients first load the scene containing the in-scene placed NetworkObject before it can be parented under the player.  As such, the server-side should receive the client's `SceneEventType.SynchronizeComplete` event before attempting to parent the in-scene placed NetworkObject under the player.

Reply:
Dang, NetworkObject parenting _is_ complicated. I had to step away and come back to this 😅 

Let me try to break this down...
- NetworkObjects you place **in-scene** spawn **after** the scene (that they are placed in) loads.
- As a result:
    - The player's client must load the scene (containing the in-scene NetworkObject) before making the player the parent of the in-scene NetworkObject.
    - As a result:
        - You should (must?) receive the client's `SceneEventType.SynchronizeComplete` or `SceneEventType.LoadComplete` event server-side before you attempt to make the player the parent of the in-scene NetworkObject.

Am I getting it?